### DISCUSSION TITLE
Nanoels project

### DISCUSSION THREAD

**Discussion by zabu83** [kachurovskiy/nanoels#6] [General]
Hello, first of all I want to thank you for sharing your project.

I'm going to start ordering all the components. I was wondering if there is any improvement to this day.

my lathe is the wm210v with 56t on the spindle, I guess I'll have to use the 56 tooth gear that another colleague shared.

What density is suitable for 3d pieces?

as I get the material I will comment.

Thank you. All the best.

**Comment by kachurovskiy** [comment]
Hi zabu83, the infill would depend on the component. For stepper mount and gears something like 50% or higher would be good. For the encoder gear and mount, 30% might be sufficient, same for the control box. You can always go higher infill if you can control/accept the warping. Cheers!

**Reply by zabu83** [reply]
Thank you very much for taking your time to respond.   

For the stepper motor assembly I had thought of plywood, can you provide me with a template to cut out? the gears I am going to opt for the belt system. do you also need 50% density? Finally, can you recommend a nema23 kit with driver? (if you don't mind purchase link) Thank you!

**Reply by kachurovskiy** [reply]
Sure, https://raw.githubusercontent.com/kachurovskiy/nanoels/main/h1/stepper/nema23-mount-plywood.pdf

Belt pulleys would need to withstand some serious force, especially one on the stepper shaft since it doesn't have a pin and the stepper will heat it up through the shaft. If you're going to run the stepper for any prolonged period of time, metal pulleys would be better. Otherwise PETG and some good wall thickness/infill might suffice.

I would recommend a closed loop Nema 23. I haven't used one (open loop is harder to make work so I use it to cover both cases) so can't recommend anything in particular. I've heard good things about Rtelligent - https://www.aliexpress.com/item/33016826807.html

Best wishes!

**Reply by zabu83** [reply]
There are 3 purchase options, which one should I buy?

**Reply by zabu83** [reply]
I have finally ordered the stepper motor, the measurement is 57 * 57 * 119, I hope that 6mm difference. Regarding the 113mm, it does not affect the mounting. 

I have seen that the shaft of the stepper motor is 8mm. Is it possible to upgrade the HDT 5M gear to 8mm? Greetings.

**Reply by kachurovskiy** [reply]
zabu83, 119mm closed loop won't fit into default case (https://raw.githubusercontent.com/kachurovskiy/nanoels/main/h1/case/case-120mm-2m6-2m5-4m3.png) but I don't recommend using the default case anyway because it leads to the constant need to disassemble the front lathe panel to check/change anything. I am myself looking to get rid of it and have the stepper driver in the electric box somewhere.

https://github.com/kachurovskiy/nanoels/raw/main/h1/stepper/htd5m-16t-8b.stl - alternatively you can get one on https://www.aliexpress.com/item/32842173382.html for a few bucks. Cheers!

**Reply by zabu83** [reply]
> zabu83, 119mm closed loop won't fit into default case (https://raw.githubusercontent.com/kachurovskiy/nanoels/main/h1/case/case-120mm-2m6-2m5-4m3.png) but I don't recommend using the default case anyway because it leads to the constant need to disassemble the front lathe panel to check/change anything. I am myself looking to get rid of it and have the stepper driver in the electric box somewhere.
> 
> https://github.com/kachurovskiy/nanoels/raw/main/h1/stepper/htd5m-16t-8b.stl - alternatively you can get one on https://www.aliexpress.com/item/32842173382.html for a few bucks. Cheers!

I guess I'll have to do some modification.
do you recommend any?

**Reply by kachurovskiy** [reply]
zabu83, I don't have an answer here yet, will be looking into the new electronics housing for my lathe as well, will let you know how that goes!

**Comment by zabu83** [comment]
Is the plywood template of the support for the stepper motor the same to make it in steel? 

do you think 5mm steel can work?

I would like to motorize the cross slider. for nema 17 engine do I need a separate driver? the code for arduino is not published? 
Thank you. 
A greeting.

**Reply by kachurovskiy** [reply]
I would suggest making it in plywood first to see if everything fits as expected, you can always go to steel later if you need to. Cutting a steel part costed me 40$ and it would be sad if it were to not fit your lathe.

5mm would be too little for steel I think because you need to make a hole through it for the tightening bolt, I think I used M4. Mine is 10mm but I guess 8mm would also work. Having thicker piece help self-aligning it at 90deg to the lead screw.

Cross slider is a whole bunch of work, you might want to get the single-axis stuff working first. Yes, it will need a separate driver. NanoELS doesn't actually support cross slide yet, it's a ton of work and chances are it might not be ready this year.

If you absolutely need the second axis you can look at www.rocketronics.de or Russian https://www.chipmaker.ru/files/file/9787/ but the latter is a hard to build because the documentation is not centralized.

**Reply by zabu83** [reply]
my lathe is exactly like yours, I'm still going to try plywood. from your explanation I think 10mm of steel is adequate. the second axis is not essential at the moment, but I would like to be able to use it at least to face. 

I have some doubts about how to connect the driver, I never worked with stepper motors. when I receive the motor and driver I will ask if you can help me. 

Your work is amazing and I thank you for your time. 

Regards.

**Reply by mockendon** [reply]
I printed mine with a pinch bolt in ASA (6 perimeters, 60% infill) and it is holding up really well.  I have had good luck making similar components out of aluminium - if the plastic mount fails I'm planning to go with that.

**Comment by zabu83** [comment]
Hi.

I have already received some parts, the motor (nema23 rtlligent brand closed loop), encoder (600p / r, power supply (48v) It is the first time that I handle with this type of engines. 

This motor carries an encoder cable to the driver, do I have to connect it to the driver? on the other hand I need to know the position of the sw1-8 buttons on the driver. 

Regards.

![Screenshot_20210418-100649_AliExpress](https://user-images.githubusercontent.com/82053978/115138763-55999380-a02e-11eb-9380-a12b1338584d.png)

**Reply by kachurovskiy** [reply]
For pulse/rev you'd want to go as low as possible - 800 - so SW1 off, SW2 on, SW3 on, SW4 on.

SW5 is something you can toggle if in the end the motor is spinning in the wrong direction. You can also toggle the direction in the Arduino code though.

SW6 - not sure what pulse filter is, I'd leave it off.

SW7 off - Arduino uses PUL/DIR scheme - one line is for providing the pusles, other is HIGH/LOW determining the direction of the motor.

SW8 on - we need this as a stepper I think.

You need to connect all the cables that came with the motor to the driver, yes.

See diagram on https://github.com/kachurovskiy/nanoels#assembly on how to connect this to the PCB.

ENA and ALM shouldn't be connected I think.

**Comment by zabu83** [comment]
Hello, I already have everything provisionally mounted and it works in the absence of further testing. 

The first problem that arises for me is that the motor from time to time stalls and starts to vibrate, to keep it running I have to go back a little and go forward again. the problem is the same with the free car. 

I have configured the sw as you told me and in arduino I have modified the motor line to 800. the sw6 button that says something about the filter, if I have it off, it vibrates all the time and doesn't move, if I turn it on it works but sometimes it locks I leave a little video for you to see.
 https://youtu.be/JFKYFiB9-04

Thank you. 
Regards.

**Reply by kachurovskiy** [reply]
Ismael, congrats, great progress! How are you sending the signal to the motor - did you already assemble the NanoEls PCB and using left-right jog buttons?

Try going through the options described in https://github.com/kachurovskiy/nanoels#my-stepper-is-skipping-steps and let me know how it goes. Thanks!

**Reply by kachurovskiy** [reply]
Also, please check the manual and instructions for your driver - http://www.rtelligent.cn/T60.html - you'd want to ensure everything is connected properly - http://www.rtelligent.cn/upload/wenjian/T60_User_Manual.pdf

**Reply by zabu83** [reply]
Thanks, the pcb board is assembled and working. 

in general everything is assembled in the absence of putting everything in its place. 

The encoder is also placed on the pcb but I still need to glue it to the lathe. To move the video, use the left and right buttons. 

the motor I am moving it with a source of 36v 5A. could that be the problem? 

I bought a 48v 5A supply but the Chinese got it wrong and sent me 36v 5A. 

Now I'm waiting for the 48v 5A power to arrive.

**Reply by kachurovskiy** [reply]
Sorry, pointed you to the wrong manual before - http://www.rtelligent.cn/upload/wenjian/T60_User_Manual.pdf

I think the problem is that I misguided you - SW8 should be off for closed loop logic to work!

**Comment by zabu83** [comment]
hehehe, don't worry. 

I realized that it was not the correct manual. I downloaded it and I was looking at all the connections and they are good. 

it is true that I realized that se8 should be off. 
I'm going to try it today when I get home from work and I'll tell you if it works. 

Thank you so much!

**Comment by zabu83** [comment]
finally everything is working as expected. 

i switched sw8 off and the problem with the motor was fixed. 
I also changed d2 and d3 for the correct left and right position. 

I'll start to do thread tests tomorrow. 
I will comment on any changes. 

Greetings.

**Comment by zabu83** [comment]
Hi, I have a doubt. 

The motor encoder cable and the geared encoder can the excess cable be cut?
![IMG_20210429_213938](https://user-images.githubusercontent.com/82053978/116608842-f2640700-a933-11eb-8cb0-fddfa417df8f.jpg)
![MVIMG_20210429_213930](https://user-images.githubusercontent.com/82053978/116608863-f859e800-a933-11eb-9be1-8d88804270ce.jpg)

**Reply by kachurovskiy** [reply]
Hey, up to you, really. I suppose if powered down, carefully stripped, checked with a multi-meter for pairs (you need to cut 2 different cables and reconnect them) and put back together it should work.

If it doesn't work then, it's going to be expensive to replace the motor.

Any chance to route the excess to the back of the machine or somewhere where it's not visible?

At the end, you're probably going to sell this stepper or use it on another project years from now and it will be much nicer if the cables are intact.

**Reply by kachurovskiy** [reply]
And for the cheap lathe spindle rotary encoder, much easier decision: I did cut mine and there were no issues. However, I'm going to have hard time moving the control board to a different location now (if I decide to re-arrange things) because the cable is too short :/

**Reply by zabu83** [reply]
you're right, if I need to make changes later I may need that cable. 

I will try to hide the cable behind the winch. 

Will the second axis update be seen soon?

**Comment by kachurovskiy** [comment]
Hey, little chances for that, not even started. You're better off getting
the Russian two-axis ELS from ChipMaker forum and translating the
instructions to English - if you ever feel like investing a few weeks into
getting the second axis built and setup.

**Comment by zabu83** [comment]
I think I will wait for your project to advance, it is very simple and effective.

**Reply by kachurovskiy** [reply]
Thank you. When your NanoELS build is done, please consider posting a little overview / walk-through video on YouTube to share your experience and help other makers considering building this too make decisions. I'd be happy to link to your video from https://github.com/kachurovskiy/nanoels/#example-builds

Many thanks!

**Reply by zabu83** [reply]
I don't think I can make a video explaining the process, especially because of the language (I only speak Spanish), but I can document it with images. 

I will open a new query here showing the process that I have followed. 
I have basically replicated everything that you have done with a few minor modifications.

ps: if you need help to test the second axis you can count on me

**Reply by kachurovskiy** [reply]
Sure, huge thanks!

**Comment by ws786873** [comment]
Hello,zabu83， can you share your PCB shell model? I also bought a motor of the same size as you. Its length is 119mm

**Reply by zabu83** [reply]
The modification was made by a friend, I will ask for the file and upload it
